src/utils/LoggedInHeader.ts

```diff
@@ -0,0 +1,59 @@
+import axios, { AxiosError } from 'axios'
+import { ErrorResponse } from '../types'
+import { useLogout } from './hooks'
+import { getProfile } from '../services/api'
+import { useCallback, useContext, useEffect } from 'react'
+import { FlashMessageType } from '../context/flashmessage'
+import { ProfileContext } from '../context/profile'
+import { TokenContext } from '../context/token'
+import { getTokenLocalStorage } from './helpers'
+
+export const useLoggedInHeader = (addErrorMessage: (flashMessage: FlashMessageType) => void): void => {
+  const { token, setToken } = useContext(TokenContext)
```

Comment:
Why the "Header" part? This could be used anywhere right?
Maybe simply useLoggedIn or something.
This hook does a few things so maybe it deserve some functions comments.
Also since this is a hook it should get in the hook directory.
Last, why do we pass the addErrorMessage in parameter, do we plan to ever use it with something different than `useContext(FlashMessageContext)`?